src/crystal/system/unix/event_loop.cr

```diff
@@ -0,0 +1,138 @@
+require "./event_libevent"
+require "./io_provider"
+
+# :nodoc:
+module Crystal::EventLoop
```

Comment:
This might be the most clear cut case of something that better be two separate classes that I've ever seen :) 

And I do mean classes and not modules, as this would also be an excellent holder of the loop specific variables (the uring and the libevent stuff, respectively). It would then be instantiated **once** and it would allow us to combat the amount of indirection somewhat.